packages/beacon-node/src/network/peers/discover.ts

```diff
@@ -245,18 +261,69 @@ export class PeerDiscovery {
     }
   }
 
+  /**
+   * Progressively called by libp2p peer discovery as a result of any query.
+   */
+  private onDiscoveredPeer = async (evt: CustomEvent<PeerInfo>): Promise<void> => {
+    const {id, multiaddrs} = evt.detail;
+    const status = await this.handleDiscoveredPeer(id, multiaddrs[0]);
+    this.metrics?.discovery.discoveredStatus.inc({status});
+  };
+  /**
+   * Progressively called by libp2p peer discovery as a result of any query.
+   */
+  private async handleDiscoveredPeer(peerId: PeerId, multiaddrTCP: Multiaddr): Promise<DiscoveredPeerStatus> {
```

Comment:
Is this repeated code from below?

Reply:
its similar. doesn't include everything but I'm not sure how to dedupe. thoughts or commits welcome